OpenTrons/opentrons-api
# feat(protocol-designer): return tip UI foundation

closes AUTH-580 AUTH-581 AUTH-582

# Overview

Decided to tackle this since I wasn't quite sure how to make the UI (since design doesn't have bandwidth rn to add this feature) so i was messing around and figured i might as well open a PR. 

What this PR does is 
- creates a ff
- creates a few new fields: `dropTipField` is modified, `pickUpTipField`, and then the well selection input field for both.

Please note that there is no error handling and the new fields are not wired up in step-generation. 

## Test Plan and Hands on Testing

- turn on the return tip ff
- create a flex or ot-2 protocol
- add 1 labware to the deck
- create a transfer step and fill out the labware selections and wells. 
- see that the pick up tip and drop tip fields are now enabled. Pick up tip should default to "default - get next tip", drop tip should default to "trash bin". 
- See that when you select tipracks for both of them, the well selection fields are revealed. Click on each well selection field and see that it opens a modal that allows you to select a well and save. There is no error handling yet so it does not error or show a ration needed. we will wire this up later!
- once you create the step, open dev tools and go to the redux tree. Navigate to `stepForms` -> `savedStepForms` -> click on the step after the initial deck setup -> navigate to the relevant fields which are `pickUpTip_location`, `dropTip_location`, `pickUpTip_wellNames` and `dropTip_wellNames` (note that you might need to expand the tree to see those keys!) See that they populate correctly with the locations showing the tiprack id and the wellnames showing a `string[]` of each well name
-  finally, repeat with a mix step. Your tiprack and well names should populate in the redux tree.

## Changelog

- create ff and put new work behind it
- create new fields and wire them up to `moveLiquid` and `mix`
- add new form data fields that are optional
- add tests to each field
- create a new selector for returning only tipracks selected
- wire up the empty form for the fields
- add a few dependent fields

## Review requests

see test plan

## Risk assessment

low, behind ff
